pytest_celery/test_services/mixins/rabbitmq.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+import pika
+from rfc3986.builder import URIBuilder
+
+from pytest_celery.compat import cached_property
+
+
+class RabbitMQTestServiceMixin:
+    def _url(self, scheme):
+        username = self._container.RABBITMQ_DEFAULT_USER
+        password = self._container.RABBITMQ_DEFAULT_PASS
+        host = self._container.get_container_host_ip()
+        port = self._container.get_exposed_port(self._container.RABBITMQ_NODE_PORT)
+
+        uri_builder = URIBuilder().add_scheme(scheme)
+        if username or password:
+            uri_builder = uri_builder.add_credentials(username, password)
+        if host:
+            uri_builder = uri_builder.add_host(host)
+        if port:
+            uri_builder = uri_builder.add_port(port)
+
+        return uri_builder.geturl()
+
+    @cached_property
+    def get_client(self) -> pika.BlockingConnection:
+        return pika.BlockingConnection(self._container.get_connection_params())
+
+    def ping(self) -> None:
+        connection = pika.BlockingConnection(self._container.get_connection_params())
+        if connection.is_open:
+            channel = connection.channel()
+            channel.queue_declare(queue="ping")
+            channel.basic_publish(exchange="", routing_key="ping", body=b"PING")
```

Comment:
This ping check is meaningless. If the connection isn't open, nothing happens. Instead, we need to raise an exception. This will happen if we try to use a channel while the connection isn't working. See [`_raise_if_not_open`](https://github.com/pika/pika/blob/11fe3772acd0ac00cc9b4a549ac630bbefb3f29d/pika/channel.py#L1376) in Pika's codebase.

```suggestion
        connection = pika.BlockingConnection(self._container.get_connection_params())
        channel = connection.channel()
        channel.queue_declare(queue="ping")
        channel.basic_publish(exchange="", routing_key="ping", body=b"PING")
```

Reply:
Now that I'm thinking about this, this is the wrong implementation for the method.

Reply:
changed to proper implementation